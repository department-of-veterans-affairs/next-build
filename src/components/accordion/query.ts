import { QueryFormatter } from 'next-drupal-query'
import { ParagraphAccordion } from '@/types/drupal/paragraph'
import { AccordionItem } from '@/components/accordion/formatted-type'
import { getHtmlFromField } from '@/lib/utils/getHtmlFromField'

export const formatter: QueryFormatter<ParagraphAccordion, AccordionItem> = (
  entity: ParagraphAccordion
) => {
  return {
    type: entity.type as AccordionItem['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id,
    header: entity.field_header || null,
    html: getHtmlFromField(entity.field_rich_wysiwyg),
  }
}
