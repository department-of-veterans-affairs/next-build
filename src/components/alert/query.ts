import { QueryFormatter } from 'next-drupal-query'
import { ParagraphAlert } from '@/types/drupal/paragraph'
import { Alert, AlertBlock, AlertType } from '@/components/alert/formatted-type'
import { formatParagraph } from '@/lib/drupal/paragraphs'
import { queries } from '@/lib/drupal/queries'

export const formatter: QueryFormatter<ParagraphAlert, Alert> = (
  entity: ParagraphAlert
) => {
  return {
    type: entity.type as Alert['type'],
    id: entity.id ?? null,
    entityId: entity.drupal_internal__id ?? null,
    alertType: entity.field_alert_type as AlertType,
    heading: entity.field_alert_heading,
    blockReference: queries.formatData(
      'block--alert',
      entity.field_alert_block_reference
    ) as AlertBlock,
    paragraphs: entity.field_va_paragraphs.map((paragraph) =>
      formatParagraph(paragraph)
    ),
  }
}
