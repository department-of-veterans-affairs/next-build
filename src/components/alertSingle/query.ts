import { QueryFormatter, QueryParams } from 'next-drupal-query'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'
import { ParagraphAlertSingle } from '@/types/drupal/paragraph'
import {
  AlertSingle,
  AlertBlock,
  AlertNonReusable,
} from '@/types/formatted/alert'
import { formatParagraph } from '@/lib/drupal/paragraphs'
import { queries } from '@/data/queries'

export const params: QueryParams<null> = () => {
  return new DrupalJsonApiParams().addInclude([
    'field_alert_block_reference',
    'field_alert_block_reference.field_alert_content',
    'field_alert_non_reusable_ref',
    'field_alert_non_reusable_ref.field_va_paragraphs',
  ])
}

export const formatter: QueryFormatter<ParagraphAlertSingle, AlertSingle> = (
  entity: ParagraphAlertSingle
) => {
  return {
    type: entity.type as AlertSingle['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id,
    alertSelection:
      entity.field_alert_selection as AlertSingle['alertSelection'],
    blockReference: queries.formatData(
      'block--alert',
      entity.field_alert_block_reference
    ) as AlertBlock,
    nonReusableRef: formatParagraph(
      entity.field_alert_non_reusable_ref
    ) as AlertNonReusable,
  }
}
