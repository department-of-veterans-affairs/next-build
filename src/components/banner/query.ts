import { QueryData, QueryFormatter } from 'next-drupal-query'
import { BANNER_RESOURCE_TYPES } from '@/lib/constants/resourceTypes'
import { BannersData } from './formatted-type'
import { drupalClient } from '@/lib/drupal/drupalClient'
import { getHtmlFromField } from '@/lib/utils/getHtmlFromField'

export type BannerDataOpts = {
  itemPath?: string
}

// The banner data endpoint is a custom endpoint provided by Drupal due to how banners are associated with a page.
// A given page does not reference a banner node via entity reference, banner node types have a field that lists what
// paths they are supposed to be visible on. This endpoint queries banners based on their path lists.
// See docroot/modules/custom/va_gov_api/src/Controller/BannerAlertsController.php in va.gov-cms for more info.
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const data: QueryData<BannerDataOpts, any> = async (opts) => {
  if (opts.itemPath) {
    const bannerUrl = `${process.env.NEXT_PUBLIC_DRUPAL_BASE_URL}/api/v1/banner-alerts?path=${opts.itemPath}`

    const response = await drupalClient.fetch(bannerUrl)
    const json = await response.json()
    return json.data
  }

  return []
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const formatter: QueryFormatter<any, BannersData> = (entities) => {
  return entities?.map((banner) => {
    switch (banner?.type.target_id as string) {
      case BANNER_RESOURCE_TYPES.BASIC:
        // this field returns 'perm' or 'dismiss' string instead of bool
        const dismiss = banner.field_dismissible_option === 'dismiss'

        return {
          id: banner.nid,
          title: banner.title,
          body: getHtmlFromField(banner.body),
          alertType: banner.field_alert_type,
          dismiss,
          type: banner.type.target_id,
        }
      case BANNER_RESOURCE_TYPES.PROMO:
        return {
          id: banner.nid,
          title: banner.title,
          href: banner.field_link?.uri,
          alertType: banner.field_promo_type,
          type: banner.type.target_id,
        }
      default:
        return null
    }
  })
}
