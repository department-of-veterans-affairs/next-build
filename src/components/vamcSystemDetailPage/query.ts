import { QueryData, QueryFormatter, QueryParams } from 'next-drupal-query'
import { NodeVamcSystemDetailPage } from '@/types/drupal/node'
import { VamcSystemDetailPage } from './formatted-type'
import { formatter as formatListOfLinkTeasers } from '@/components/listOfLinkTeasers/query'
import {
  PARAGRAPH_RESOURCE_TYPES,
  RESOURCE_TYPES,
} from '@/lib/constants/resourceTypes'
import { ExpandedStaticPropsContext } from '@/lib/drupal/staticProps'
import {
  entityBaseFields,
  fetchSingleEntityOrPreview,
  getMenu,
} from '@/lib/drupal/query'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'
import { Menu } from '@/types/drupal/menu'
import { buildSideNavDataFromMenu } from '@/lib/drupal/facilitySideNav'
import {
  getLovellVariantOfBreadcrumbs,
  getLovellVariantOfUrl,
  getOppositeChildVariant,
} from '@/lib/drupal/lovell/utils'
import { formatter as formatAdministration } from '@/components/administration/query'
import { drupalClient } from '@/lib/drupal/drupalClient'
import { formatParagraph } from '@/lib/drupal/paragraphs'
import { getNestedIncludes } from '@/lib/utils/queries'

export const params: QueryParams<null> = () => {
  return new DrupalJsonApiParams().addInclude([
    'field_administration',
    'field_office',
    'field_related_links',
    'field_related_links.field_va_paragraphs',
    ...getNestedIncludes('field_featured_content', PARAGRAPH_RESOURCE_TYPES.QA),
  ])
  // I would like to be able to use just these recursive fields, but it doesn't seem to
  // work, at least with this version of Drupal. According to the documentation here
  // https://www.drupal.org/docs/core-modules-and-themes/core-modules/jsonapi-module/fetching-resources-get#s-get-article-media-entity-reference-field-image-url-uri-by-including-references
  // there's an "older syntax" and "new syntax" for the query string this produces.
  // .addFields('paragraph--list_of_link_teasers', ['field_va_paragraphs'])
  // .addFields('paragraph--collapsible_panel', ['field_va_paragraphs'])
  // .addFields('paragraph--q_a', ['field_answer'])
}

// Define the option types for the data loader.
export type VamcSystemDetailPageDataOpts = {
  id: string
  context?: ExpandedStaticPropsContext
}

// Define the data structure returned from the query.
type VamcSystemDetailPageData = {
  entity: NodeVamcSystemDetailPage
  menu: Menu | null
  hasLovellCounterpart: boolean
  lovell?: ExpandedStaticPropsContext['lovell']
}

// Implement the data loader.
export const data: QueryData<
  VamcSystemDetailPageDataOpts,
  VamcSystemDetailPageData
> = async (opts): Promise<VamcSystemDetailPageData> => {
  const entity = (await fetchSingleEntityOrPreview(
    opts,
    RESOURCE_TYPES.VAMC_SYSTEM_DETAIL_PAGE,
    params
  )) as NodeVamcSystemDetailPage

  if (!entity) {
    throw new Error(
      `NodeVamcSystemDetailPage entity not found for id: ${opts.id}`
    )
  }

  // TODO: There seems to be some difference between a "facilitySidebar" and "outreachSidebar" in the original template

  // Determine if there is a Lovell counterpart to this page so we can show or hide the Lovell switcher
  let hasLovellCounterpart = false
  if (opts.context?.lovell?.isLovellVariantPage) {
    try {
      const counterpartPath = getLovellVariantOfUrl(
        entity.path.alias,
        getOppositeChildVariant(opts.context?.lovell?.variant)
      )
      hasLovellCounterpart =
        (await drupalClient.translatePath(counterpartPath)) !== null
    } catch (error) {
      // If we're using proxy-fetcher, it'll actually throw an error for these
      if ([404, 403].includes(error.cause?.status)) {
        hasLovellCounterpart = false
      } else {
        throw error
      }
    }
  }

  // Fetch the menu name dynamically off of the field_region_page reference if available.
  const menu = await getMenu(
    entity.field_office.field_system_menu.resourceIdObjMeta
      .drupal_internal__target_id
  )

  return { entity, menu, lovell: opts.context?.lovell, hasLovellCounterpart }
}

export const formatter: QueryFormatter<
  VamcSystemDetailPageData,
  VamcSystemDetailPage
> = ({ entity, menu, lovell, hasLovellCounterpart }) => {
  // For this particular content type, which can be bifurcated, the entity path doesn't
  // always match the path of the page; it could be the opposite Lovell variant's path.
  const normalizedPath = lovell?.isLovellVariantPage
    ? getLovellVariantOfUrl(entity.path.alias, lovell.variant)
    : entity.path.alias

  return {
    ...entityBaseFields(entity),
    breadcrumbs: lovell?.isLovellVariantPage
      ? getLovellVariantOfBreadcrumbs(entity.breadcrumbs, lovell.variant)
      : entity.breadcrumbs,
    title: entity.title,
    path: normalizedPath,
    introText: entity.field_intro_text,
    showTableOfContents: entity.field_table_of_contents_boolean,
    menu: buildSideNavDataFromMenu(normalizedPath, menu),
    administration: formatAdministration(entity.field_administration),
    vamcEhrSystem: entity.field_office?.field_vamc_ehr_system || null,
    vamcSystem: {
      path: entity.field_office?.path.alias || null,
    },
    featuredContent:
      entity.field_featured_content?.map((paragraph) =>
        formatParagraph(paragraph)
      ) || null,
    relatedLinks: entity.field_related_links
      ? formatListOfLinkTeasers(entity.field_related_links)
      : null,
    lovellVariant: lovell?.variant ?? null,
    lovellSwitchPath: lovell?.isLovellVariantPage
      ? getLovellVariantOfUrl(
          normalizedPath,
          getOppositeChildVariant(lovell?.variant)
        )
      : null,
    showLovellSwitcher: hasLovellCounterpart,
  }
}
