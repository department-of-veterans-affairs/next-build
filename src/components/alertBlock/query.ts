// Define the query params for fetching block--alert.
import { BlockAlert } from '@/types/drupal/block'
import { QueryFormatter, QueryParams } from 'next-drupal-query'
import { AlertBlock, AlertType } from '@/components/alert/formatted-type'
import { formatParagraph } from '@/lib/drupal/paragraphs'
import { Wysiwyg } from '@/components/wysiwyg/formatted-type'
import { ExpandableText } from '@/components/expandableText/formatted-type'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'

export const params: QueryParams<null> = () =>
  new DrupalJsonApiParams().addInclude(['field_alert_content'])

export const formatter: QueryFormatter<BlockAlert, AlertBlock> = (
  entity: BlockAlert
) => {
  if (!entity) {
    return null
  }

  return {
    alertType: (entity.field_alert_type === 'information'
      ? 'info'
      : entity.field_alert_type) as AlertType,
    id: entity.id,
    title: entity.field_alert_title,
    content: formatParagraph(entity.field_alert_content) as
      | Wysiwyg
      | ExpandableText,
    // {
    //   header:
    //     entity.field_alert_content.field_text_expander ??
    //     entity.field_alert_heading,
    //   text: entity.field_alert_content?.field_wysiwyg?.processed ?? '',
    // },
  }
}
