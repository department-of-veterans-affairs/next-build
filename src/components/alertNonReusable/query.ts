// Define the query params for fetching block--alert.
import { QueryFormatter } from 'next-drupal-query'
import { AlertType, AlertNonReusable } from '@/components/alert/formatted-type'
import { ParagraphNonReusableAlert } from '@/types/drupal/paragraph'
import { formatParagraph } from '@/lib/drupal/paragraphs'

export const formatter: QueryFormatter<
  ParagraphNonReusableAlert,
  AlertNonReusable
> = (entity: ParagraphNonReusableAlert) => {
  if (!entity) {
    return null
  }

  return {
    type: entity.type as AlertNonReusable['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id,
    alertType: (entity.field_alert_type === 'information'
      ? 'info'
      : entity.field_alert_type) as AlertType,
    heading: entity.field_alert_heading,
    paragraphs: entity.field_va_paragraphs.map((paragraph) => formatParagraph(paragraph)),
  }
}
