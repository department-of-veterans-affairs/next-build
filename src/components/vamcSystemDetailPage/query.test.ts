/**
 * @jest-environment node
 */

import mockPage from './mock.json'
import mockMenu from './mock.menu.json'
import { queries } from '@/lib/drupal/queries'
import { RESOURCE_TYPES } from '@/lib/constants/resourceTypes'
import { VamcSystemDetailPageDataOpts } from './query'

const mockPageQuery = jest.fn(() => mockPage)

jest.mock('@/lib/drupal/query', () => ({
  ...jest.requireActual('@/lib/drupal/query'),
  fetchSingleEntityOrPreview: () => mockPageQuery(),
  getMenu: () => mockMenu,
}))

function runQuery(options: Partial<VamcSystemDetailPageDataOpts> = {}) {
  return queries.getData(RESOURCE_TYPES.VAMC_SYSTEM_DETAIL_PAGE, {
    id: mockPage.id,
    ...options,
  })
}

describe('VamcSystemDetailPage query module', () => {
  beforeEach(() => {
    // Reset to default mock data before each test
    mockPageQuery.mockReturnValue(mockPage)
  })

  test('outputs formatted data', async () => {
    expect(await runQuery()).toMatchSnapshot()
  })

  describe('handles null fields', () => {
    test('handles null introText', async () => {
      mockPageQuery.mockReturnValue({
        ...mockPage,
        field_intro_text: null,
      })

      const result = await runQuery()

      expect(result.introText).toBeNull()
    })

    test('handles null showTableOfContents', async () => {
      mockPageQuery.mockReturnValue({
        ...mockPage,
        field_table_of_contents_boolean: null,
      })

      const result = await runQuery()

      expect(result.showTableOfContents).toBeNull()
    })

    test('handles null relatedLinks', async () => {
      mockPageQuery.mockReturnValue({
        ...mockPage,
        field_related_links: null,
      })

      const result = await runQuery()

      expect(result.relatedLinks).toBeNull()
    })

    test('handles null breadcrumbs', async () => {
      mockPageQuery.mockReturnValue({
        ...mockPage,
        breadcrumbs: null,
      })

      const result = await runQuery()

      expect(result.breadcrumbs).toBeNull()
    })

    test('handles null lovell context', async () => {
      const result = await runQuery({
        context: {
          path: '/test-path',
          drupalPath: '/test-drupal-path',
          listing: {
            isListingPage: false,
            firstPagePath: '/test-first-page-path',
            page: 1,
          },
          lovell: null,
        },
      })

      expect(result.lovellVariant).toBeNull()
      expect(result.lovellSwitchPath).toBeNull()
      expect(result.breadcrumbs).toEqual(mockPage.breadcrumbs)
    })
  })
})
