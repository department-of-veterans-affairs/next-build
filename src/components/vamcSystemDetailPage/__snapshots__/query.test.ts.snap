// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`VamcSystemDetailPage query module outputs formatted data 1`] = `
{
  "administration": {
    "entityId": 218,
    "name": "VA Durham health care",
  },
  "breadcrumbs": [
    {
      "options": [],
      "title": "Home",
      "uri": "https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/",
    },
    {
      "options": [],
      "title": "VA Durham health care",
      "uri": "https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/durham-health-care",
    },
    {
      "options": [],
      "title": "ABOUT VA DURHAM",
      "uri": "internal:#",
    },
    {
      "options": [],
      "title": "About us",
      "uri": "https://content-build-medc0xjkxm4jmpzxl3tfbcs7qcddsivh.ci.cms.va.gov/durham-health-care/about-us",
    },
    {
      "options": [],
      "title": "Mission and vision",
      "uri": "internal:#",
    },
  ],
  "entityId": 6983,
  "entityPath": "/durham-health-care/about-us/mission-and-vision",
  "featuredContent": [
    {
      "html": "<p>Beginning Sept. 6 and continuing through November, our 'Blue Canopy' entrance to Bldgs. 133CA and 134 will be closed. The adjacent parking lot will also be closed. The Metra train shuttle stop at the 'Blue Canopy' entrance will be closed starting Friday, Sept. 5, 2025. Please instead use the shuttle stop at the entrance of Bldg. 131.</p>",
      "id": "55e4001e-1727-4c9c-970d-f985da93dfa8",
      "type": "paragraph--wysiwyg",
    },
  ],
  "id": "6f66f92d-4e84-476b-83fb-0457ce0ab4ae",
  "introText": "The mission of the Durham VA Medical Center and clinics is to make a positive difference in the lives of the Veterans we serve by providing compassionate, comprehensive, and quality healthcare.  The Durham VA Medical Center and clinics promote a community of integrity, respect, stewardship, and excellence by understanding and respecting the diversity of those whom we serve.",
  "lastUpdated": "2021-03-18T15:01:54+00:00",
  "lovellSwitchPath": null,
  "lovellVariant": null,
  "mainContent": [
    {
      "html": "<h4><strong>Psychology Doctoral Internship</strong></h4><p>Application Deadline: November 1</p><p>The APA accredited internship at the Lexington VAHCS is a full-time, generalist internship in clinical psychology. The goal of our program is to prepare trainees for clinical practice of psychology in a variety of settings with an emphasis on preparation for VA and other medical settings. Our program offers a wide breadth of clinical training opportunities focused on evidence-based practice. Interns generally select their own rotations that fit their training goals. We pride ourselves on creating a friendly, nurturing training environment and aim to create a positive, collegial atmosphere that is conducive for learning and growth.</p><p> </p>",
      "id": "5fa9eeac-1089-46da-8ecf-b25562a10f24",
      "type": "paragraph--wysiwyg",
    },
    {
      "entityId": 112056,
      "extension": "PDF",
      "id": "fd014675-7b4c-482b-9167-5389f1dc44ce",
      "mediaType": "document",
      "title": "Lexington VA Internship Admissions, Support and Initial Placement Data",
      "type": "paragraph--downloadable_file",
      "url": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/2024-08/Internship%20Admissions%20Support%20and%20Inital%20Placement%20Data.pdf",
    },
    {
      "entityId": 112057,
      "extension": "PDF",
      "id": "54328865-016b-4694-8bb2-45144c8a61cb",
      "mediaType": "document",
      "title": "Lexington VA Psychology Internship Brochure 2026 - 2027",
      "type": "paragraph--downloadable_file",
      "url": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/2025-09/Lexington%20Psychology%20Internship%20Brochure%202026-2027.pdf",
    },
    {
      "body": "<p>Robert W. Askey, MBA, AAS CA, is the Associate Director of the Corporal Michael J. Crescenz VA Medical Center (CMCVAMC). A  retired colonel of the United States Army,  he served in a variety of transportation and logistics assignments both stateside and overseas, to include a tour on the Army General Staff. Mr. Askey began his career with the VA in 2014 when he joined the  Voluntary Service Team in support of the National Veterans Wheelchair Games, hosted that year by the Philadelphia VA Medical Center. His next  assignment with the CMCVAMC was on the Logistics Service leadership team. Mr. Askey then relocated to North Carolina to serve as the Assistant Chief of Logistics at the Durham VA Healthcare System. In 2017, he returned to the CMCVAMC as the Chief Supply Chain Officer. Mr. Askey earned his Master’s Degree in Business Administration from the Florida Institute of Technology, and his Culinary Degree from the Culinary Institute of Virginia.</p>",
      "completeBiographyCreate": true,
      "description": "Associate Director",
      "emailAddress": null,
      "entityId": 182998,
      "firstName": "Bob",
      "id": "5d77166a-ed09-400c-b858-d7798cd12f04",
      "introText": "Robert W. Askey, MBA, AAS CA, is the Associate Director of the VA Philadelphia health care system. ",
      "lastName": "Askey",
      "media": {
        "alt": "Photo of Bob Askey, Acting associate director for support services",
        "height": 136,
        "id": "ac356195-9631-43c0-a0fc-d84f4ea362eb",
        "links": {
          "1_1_square_medium_thumbnail": {
            "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/1_1_square_medium_thumbnail/public/2020-04/Bob_Askey.jpg",
            "meta": {
              "linkParams": {
                "rel": [
                  "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
                ],
                "width": "240",
              },
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
              ],
            },
          },
          "2_1_large": {
            "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/2_1_large/public/2020-04/Bob_Askey.jpg",
            "meta": {
              "linkParams": {
                "rel": [
                  "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
                ],
                "width": "512",
              },
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
              ],
            },
          },
          "2_1_medium_thumbnail": {
            "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/2_1_medium_thumbnail/public/2020-04/Bob_Askey.jpg",
            "meta": {
              "linkParams": {
                "rel": [
                  "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
                ],
                "width": "240",
              },
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
              ],
            },
          },
          "2_3_medium_thumbnail": {
            "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/2_3_medium_thumbnail/public/2020-04/Bob_Askey.jpg",
            "meta": {
              "linkParams": {
                "rel": [
                  "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
                ],
                "width": "480",
              },
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
              ],
            },
          },
          "3_2_medium_thumbnail": {
            "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/3_2_medium_thumbnail/public/2020-04/Bob_Askey.jpg",
            "meta": {
              "linkParams": {
                "rel": [
                  "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
                ],
                "width": "320",
              },
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
              ],
            },
          },
          "7_2_medium_thumbnail": {
            "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/7_2_medium_thumbnail/public/2020-04/Bob_Askey.jpg",
            "meta": {
              "linkParams": {
                "rel": [
                  "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
                ],
                "width": "300",
              },
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
              ],
            },
          },
          "full_content_width": {
            "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/full_content_width/public/2020-04/Bob_Askey.jpg",
            "meta": {
              "linkParams": {
                "rel": [
                  "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
                ],
              },
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
              ],
            },
          },
          "large": {
            "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/large/public/2020-04/Bob_Askey.jpg",
            "meta": {
              "linkParams": {
                "rel": [
                  "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
                ],
              },
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
              ],
            },
          },
          "original": {
            "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/original/public/2020-04/Bob_Askey.jpg",
            "meta": {
              "linkParams": {
                "rel": [
                  "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
                ],
              },
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
              ],
            },
          },
          "self": {
            "href": "http://va-gov-cms.ddev.site/jsonapi/file/file/ac356195-9631-43c0-a0fc-d84f4ea362eb",
          },
          "viewport_width": {
            "href": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/styles/viewport_width/public/2020-04/Bob_Askey.jpg",
            "meta": {
              "linkParams": {
                "rel": [
                  "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
                ],
              },
              "rel": [
                "drupal://jsonapi/extensions/consumer_image_styles/links/relation-types/#derivative",
              ],
            },
          },
        },
        "title": "",
        "type": "media--image",
        "width": 110,
      },
      "phoneNumber": null,
      "photoAllowHiresDownload": false,
      "suffix": "MBA, AAS CA",
      "type": "paragraph--staff_profile",
      "vamcTitle": null,
    },
    {
      "displayAccordion": false,
      "header": "Q & A group header",
      "id": "0029a20f-5757-4053-9820-3db039bfa3c7",
      "intro": "This is some intro text",
      "questions": [
        {
          "answers": [
            {
              "html": "<p>Answer with some extras</p>",
              "id": "2d640021-2068-4233-8651-8fc4de79b846",
              "type": "paragraph--wysiwyg",
            },
            {
              "data": [
                [
                  "a",
                  "b",
                ],
                [
                  "c",
                  "d",
                ],
                [
                  "e",
                  "f",
                ],
              ],
              "entityId": 183000,
              "id": "710639b1-353f-4583-81a4-2b1a41de5f82",
              "title": "I'm a table nested in a Q&A",
              "type": "paragraph--table",
            },
            {
              "description": "<p>Every 2 is a loser</p>",
              "entityId": 183001,
              "id": "e5a962fa-e751-4b54-a9f5-777548618bfe",
              "numberPhrase": "Every 1 is a winner",
              "type": "paragraph--number_callout",
            },
          ],
          "id": "211ce79f-5876-4d76-b848-b26532d941a1",
          "question": "Question 1",
          "type": "paragraph--q_a",
        },
      ],
      "type": "paragraph--q_a_section",
    },
    {
      "entityId": 183005,
      "id": "f5cedf99-5295-4877-901f-223d92e369d3",
      "linkTeasers": [
        {
          "componentParams": {
            "sectionHeader": "",
          },
          "entityId": 183004,
          "id": "8db029e2-8f65-40ff-9f1a-cb726814f346",
          "options": {
            "data-entity-substitution": "",
            "data-entity-type": "",
            "data-entity-uuid": "",
            "href": "https://www.google.com/",
          },
          "parentField": "field_va_paragraphs",
          "summary": "This is a summary of the Google link",
          "title": "Google",
          "type": "paragraph--link_teaser",
          "uri": "https://www.google.com/",
        },
      ],
      "title": "Link teasers",
      "type": "paragraph--list_of_link_teasers",
    },
    {
      "bordered": false,
      "entityId": 183008,
      "id": "0a4daab1-c755-401c-bc01-82488a4370e6",
      "paragraphs": [
        {
          "entityId": 183007,
          "id": "1f1a794b-d061-4112-8c45-e3c6daa5b647",
          "paragraphs": [
            {
              "data": [
                [
                  "11",
                  "12",
                ],
                [
                  "21",
                  "22",
                ],
                [
                  "31",
                  "32",
                ],
                [
                  "",
                  "",
                ],
                [
                  "",
                  "",
                ],
              ],
              "entityId": 183006,
              "id": "662f086d-e329-4978-8771-fd9a7c22c832",
              "title": "A table inside a root-level accordion item",
              "type": "paragraph--table",
            },
          ],
          "title": "A root-level accordion item",
          "type": "paragraph--collapsible_panel_item",
          "wysiwyg": "<p>Some text inside</p>",
        },
      ],
      "type": "paragraph--collapsible_panel",
    },
    {
      "allowClicksOnThisImage": true,
      "entityId": 183009,
      "id": "bc5d91c0-ae80-4b24-a942-e2c3688bbe8d",
      "image": {
        "alt": "Rating sandwiches",
        "title": "",
        "url": "https://dsva-vagov-staging-cms-files.s3.us-gov-west-1.amazonaws.com/2025-10/my-image.png",
      },
      "type": "paragraph--media",
    },
  ],
  "menu": {
    "data": {
      "description": null,
      "links": [
        {
          "description": null,
          "expanded": false,
          "label": "VA Bronx health care",
          "links": [
            {
              "description": null,
              "expanded": false,
              "label": "SERVICES AND LOCATIONS",
              "links": [],
              "lovellSection": null,
              "url": {
                "path": "",
              },
            },
            {
              "description": null,
              "expanded": false,
              "label": "NEWS AND EVENTS",
              "links": [],
              "lovellSection": null,
              "url": {
                "path": "",
              },
            },
            {
              "description": null,
              "expanded": false,
              "label": "ABOUT VA RICHMOND",
              "links": [],
              "lovellSection": null,
              "url": {
                "path": "",
              },
            },
          ],
          "lovellSection": null,
          "url": {
            "path": "/louisville-health-care",
          },
        },
      ],
      "name": "VA Bronx health care",
    },
    "rootPath": "/durham-health-care/about-us/mission-and-vision/",
  },
  "metatags": [],
  "moderationState": "published",
  "path": "/durham-health-care/about-us/mission-and-vision",
  "published": true,
  "relatedLinks": {
    "entityId": 19369,
    "id": "928ec63a-97b3-460d-a869-1d0b1ca9fa04",
    "linkTeasers": [
      {
        "componentParams": {
          "sectionHeader": "",
        },
        "entityId": 19367,
        "id": "95f2a5f6-82b3-447d-afa3-be1a4560fa03",
        "options": [],
        "parentField": "field_va_paragraphs",
        "summary": "Learn more about the VA Durham leadership team.",
        "title": "Our leadership",
        "type": "paragraph--link_teaser",
        "uri": "entity:node/6226",
      },
      {
        "componentParams": {
          "sectionHeader": "",
        },
        "entityId": 19368,
        "id": "2ef38bb5-0c45-4ba6-bd02-e1900880bb4a",
        "options": [],
        "parentField": "field_va_paragraphs",
        "summary": "Find health care and services you need to stay healthy.",
        "title": "Health services",
        "type": "paragraph--link_teaser",
        "uri": "entity:node/6962",
      },
    ],
    "title": "Learn more about how we serve Veterans",
    "type": "paragraph--list_of_link_teasers",
  },
  "showLovellSwitcher": false,
  "showTableOfContents": true,
  "title": "Mission and vision",
  "type": "node--health_care_region_detail_page",
  "vamcEhrSystem": "vista",
  "vamcSystem": {
    "path": "/durham-health-care",
  },
}
`;
